### PR TITLE
Move test support files into tests directory

### PR DESCRIPTION
**Pull request by thomwiggers**
This should prevent the kind of confusion that happend in #199.

### CODE REVIEW

**Comment by dstebila** [comment]
@thomwiggers Is the Travis failure here the same thing that you talked about in #201?

**Comment by thomwiggers** [comment]
Unsure, I think it disappeared from the history while I was on vacation.

**Comment by thomwiggers** [comment]
Either this PR or #206 will need to be rebased when the other gets merged.

File: test/Makefile [hunk 1]
```diff
@@ -47,34 +48,34 @@ printparams: $(DEST_DIR)/printparams_$(SCHEME)_$(IMPLEMENTATION)
...
 	$(CC) $(CFLAGS) $< $(COMMON_FILES) -o $@
-$(DEST_DIR)/test_common_sp800-185: common/sp800-185.c $(COMMON_FILES)
+$(DEST_DIR)/test_common_sp800-185: test_common/sp800-185.c $(COMMON_FILES)
 	mkdir -p $(DEST_DIR)
 	$(CC) $(CFLAGS) $< $(COMMON_FILES) -o $@
 
 $(DEST_DIR)/functest_$(SCHEME)_$(IMPLEMENTATION): build-scheme crypto_$(TYPE)/functest.c $(COMMON_FILES) $(COMMON_DIR)/randombytes.c $(COMMON_HEADERS)
 	mkdir -p $(DEST_DIR)
-	$(CC) $(CFLAGS) -DNTESTS=$(NTESTS) -DPQCLEAN_NAMESPACE=PQCLEAN_$(SCHEME_UPPERCASE)_$(IMPLEMENTATION_UPPERCASE) -I$(SCHEME_DIR) crypto_$(TYPE)/functest.c $(COMMON_FILES) $(COMMON_DIR)/notrandombytes.c -o $@ -L$(SCHEME_DIR) -l$(SCHEME)_$(IMPLEMENTATION)
+	$(CC) $(CFLAGS) -DNTESTS=$(NTESTS) -DPQCLEAN_NAMESPACE=PQCLEAN_$(SCHEME_UPPERCASE)_$(IMPLEMENTATION_UPPERCASE) -I$(SCHEME_DIR) crypto_$(TYPE)/functest.c $(COMMON_FILES) $(COMMON_DIR)/randombytes.c -o $@ -L$(SCHEME_DIR) -l$(SCHEME)_$(IMPLEMENTATION)
```
[the remark this answers is not in the document]
Comment (thomwiggers) [reply test/Makefile]: I thought that this would benefit from having functest not test the same thing each time.

**Pull request merged**